packages/core/contracts/trader/uniswap-broker/UniswapBroker.sol

```diff
@@ -23,57 +23,56 @@ contract UniswapBroker {
      * possible to the truePrice.
      * @dev True price is expressed in the ratio of token A to token B.
      * @dev The caller must approve this contract to spend whichever token is intended to be swapped.
+     * @param tradingAsEOA bool to indicate if the UniswapBroker is being called by a DSProxy or an EOA.
```

Comment:
few small changes the the `UniswapBroker`. The two main changes are:
1) group common vars into arrays (i.e `tokenA` & `tokenB`) are joined together to form an array of addresses.
2) add `tradingAsEOA` that adds an extra token transfer that is only required when using EOA to enable DSProxy support in this module.

Reply:
Does 1 impact gas at all?